Source/Core/Core/IOS/Network/KD/Mail/WC24Send.h

```diff
@@ -31,11 +33,55 @@ class WC24SendList final
   void WriteSendList() const;
 
   std::string_view GetMailFlag() const;
+  u32 GetNumberOfMail() const;
+  std::vector<u32> GetMailToSend() const;
+  u32 GetEntryId(u32 entry_index) const;
+  u32 GetMailSize(u32 index) const;
+  ErrorCode DeleteMessage(u32 index);
+  std::string GetMailPath(u32 index) const;
+  u32 GetNextEntryId() const;
+  u32 GetNextEntryIndex() const;
+
+  ErrorCode AddRegistrationMessages(const WC24FriendList& friend_list, u64 sender);
 
 private:
   static constexpr u32 MAX_ENTRIES = 127;
   static constexpr u32 SEND_LIST_SIZE = 16384;
 
+  // Format for the message Wii Mail sends when trying to register a Wii Friend.
+  // Most fields can be static such as the X-Wii-AppId which is the Wii Menu,
+  // X-Wii-Cmd which is the registration command, and the attached file which is
+  // just 128 bytes of base64 encoded 0 bytes. That file is supposed to be friend profile data which
+  // is written to nwc24fl.bin, although it has been observed to always be 0.
+  static constexpr char MAIL_REGISTRATION_STRING[] =
+      "MAIL FROM: {0:016d}@wii.com\r\n"
+      "RCPT TO: {1:016d}wii.com\r\n"
+      "DATA\r\n"
+      "Date: {2:%a, %d %b %Y %X} GMT\r\n"
+      "From: {0:016d}@wii.com\r\n"
+      "To: {1:016d}@wii.com\r\n"
+      "Message-Id: <00002000B0DF6BB47FE0303E0DB0D@wii.com>\r\n"
```

Comment:
I am wondering if I should generate this, my main argument against is that it is never validated.

The message id formed like this
```c
sprintf(buf, "%05X%016LX%08X", msg_id, user_id, minutes_since_1900)
```